src/index.js

```diff
@@ -264,6 +275,40 @@ class IpfsRepo {
     }
   }
 
+  async _migrate (toVersion) {
+    let disableMigrationsConfig
+    try {
+      disableMigrationsConfig = await this.config.get('repoDisableAutoMigration')
+    } catch (e) {
+      if (e.code === ERRORS.NotFoundError.code) {
+        disableMigrationsConfig = false
+      } else {
+        throw e
+      }
+    }
+
+    if (disableMigrationsConfig) {
+      throw new ERRORS.InvalidRepoVersionError('Incompatible repo versions. Automatic migrations disabled. Please migrate the repo manually.')
+    }
+
+    const currentRepoVersion = await this.version.get()
+    log(currentRepoVersion)
+    if (currentRepoVersion >= toVersion) {
+      if (currentRepoVersion > toVersion) {
+        log('Your repo\'s version is higher then this version of js-ipfs-repo require! You should revert it.')
+      }
+
+      log('Nothing to migrate')
+      return
+    }
```

Comment:
I'm not sure I understand this - I think this check needs to be done in the migrator.

If `currentRepoVersion` === `toVersion` then nothing to do, no error.
If `currentRepoVersion` > `toVersion` then it needs downgrade, error if no downgrade path.

Reply:
Yeah, this definitely can be more streamlined. 

Sooo you are saying that it should automatically also downgrade? This is currently not happening and from my experience with migration tools, it is also usually not the case. There often is a way how to downgrade (and our migrator supports it), but it is not automatically integrated. I think that the main reason is to provide easy way "forward", but way "back" is usually something that most users won't need.

I checked `go` and they don't supports it.

Reply:
True about the browsers.

To your question, yes we can, but not always. Some migrations will be irreversible and in that case, an exception will be the answer I guess.